src/github.com/getlantern/flashlight/genconfig/checkfallbacks.go

```diff
@@ -0,0 +1,166 @@
+	}
+
+	numcores := runtime.NumCPU()
+
+	if *verbose {
+		log.Debugf("Using all %d cores on machine", numcores)
+	}
+	runtime.GOMAXPROCS(numcores)
+
+	fallbacks := make(chan fallbackServer)
+	loadFallbacks(fallbacks)
+	testFallbacks(fallbacks)
+}
+
+// Load the fallback servers list file. Failure to do so will result in
+// exiting the program.
+func loadFallbacks(fallbacks chan<- fallbackServer) {
+	if *fallbacksFile == "" {
+		log.Error("Please specify a fallbacks file")
+		flag.Usage()
+		os.Exit(2)
+	}
+
+	fallbacksBytes, err := ioutil.ReadFile(*fallbacksFile)
+	if err != nil {
+		log.Fatalf("Unable to read fallbacks file at %s: %s", *fallbacksFile, err)
+		os.Exit(2)
+	}
+
+	var unmarshalledFallbacks []fallbackServer
+	err = json.Unmarshal(fallbacksBytes, &unmarshalledFallbacks)
+	if err != nil {
+		log.Fatalf("Unable to unmarshal json from %v: %v", *fallbacksFile, err)
+		os.Exit(1)
+	}
+
+	// Channel fallback servers on-demand
+	go func() {
+		for _, val := range unmarshalledFallbacks {
+			fallbacks <- val
+		}
+		close(fallbacks)
+	}()
+}
+
+// Test fallback servers. Output to stdout *only* when a test fails.
+func testFallbacks(fallbacks <-chan fallbackServer) {
+	workersWg := sync.WaitGroup{}
+
+	if *verbose {
+		fmt.Printf("Spawning %d workers\n", *numConns)
+	}
+
+	workersWg.Add(*numConns)
+	for i := 0; i < *numConns; i++ {
+		// Worker: consume fallback servers from channel and signal
+		// Done() when closed (i.e. range exits)
+		go func(i int) {
+			for fb := range fallbacks {
+				testFallbackServer(fb, i)
+			}
+			workersWg.Done()
+		}(i+1)
+	}
+
+	workersWg.Wait()
+}
+
+// Perform the test of an individual fallbackServer
+func testFallbackServer(fb fallbackServer, workerId int) {
+	ip := fb["ip"].(string)
+	if fb["pt"] != nil {
+		fmt.Printf("Skipping fallback %v because it has pluggable transport enabled", ip)
+		return
+	}
+
+	cert := fb["cert"].(string)
+	// Replace newlines in cert with newline literals
+	fb["cert"] = strings.Replace(cert, "\n", "\\n", -1)
+
+	// Test connectivity
+	info := &client.ChainedServerInfo{
+		Addr:      ip + ":443",
+		Cert:      cert,
+		AuthToken: fb["auth_token"].(string),
+		Pipelined: true,
+	}
+	dialer, err := info.Dialer()
+	if err != nil {
+		fmt.Printf("%v: error building dialer: %v", ip, err)
+		return
+	}
+	c := &http.Client{
+		Transport: &http.Transport{
+			Dial: dialer.Dial,
+		},
+	}
+	req, err := http.NewRequest("GET", "http://www.google.com/humans.txt", nil)
+	resp, err := c.Do(req)
+	if err != nil {
+		fmt.Printf("%v: requesting humans.txt failed: %v", ip, err)
+		return
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != 200 {
+		fmt.Printf("%v: bad status code: %v", ip, resp.StatusCode)
+		return
+	}
+	bytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		fmt.Printf("%v: error reading response body: %v", ip, err)
+		return
+	}
+	body := string(bytes)
+	if body != expectedBody {
+		fmt.Printf("%v: wrong body: %s", ip, body)
+		return
+	}
+
+	if *verbose {
```

Comment:
Again, I think you can just log this out and forget about the `*verbose` flag altogether.  I don't remember if the verbose flag was my idea; if so, sorry. :)  I didn't know that log output went to stderr.